### PR TITLE
Child Gauge Factory

### PR DESCRIPTION
**Pull request by bout3fiddy**
Adds child gauge factory to the deployment process.

Child gauges exist on layer2s or non-ethereum layer1s (evm) and facilitate incentives for LPs.

for transferring CRV incentives from L1 to L2, the dao needs to intervene. This is a very minimal version where we can have gauges which do not emit CRV emissions, but allow rewards.

CRV emissions can always come later.

### CODE REVIEW

File: contracts/amm/stableswap/factory/factory_v_100.vy [hunk 1]
```diff
@@ -110,19 +112,31 @@ fee_receiver: public(address)
...
 
 
+@external
+def set_owner(_owner: address):
+    
+    assert msg.sender == deployer
+    assert self.admin == deployer
+    assert _owner != deployer
+
+    self.admin = _owner
```
Comment (amfet42) [inline contracts/amm/stableswap/factory/factory_v_100.vy:137]: Maybe we want an ability to change owners more than 1 time? For example, we might update DAO contract

File: scripts/deploy/__init__.py [hunk 2]
```diff
@@ -130,9 +113,52 @@ def run_deploy_all(chain: str) -> None:
...
-    update_metaregistry(chain, metaregistry, address_provider)
-
-    # TODO: transfer ownership to dao
+    update_metaregistry(chain_settings, metaregistry, address_provider)
+
+    # transfer ownership to the dao
+    owner = chain_settings.dao.ownership_admin
+
+    # addressprovider
+    current_owner = address_provider._storage.admin.get()
```
Comment (amfet42) [inline scripts/deploy/__init__.py:122]: better do `.admin()` everywhere

File: scripts/deploy/gauge/child_gauge.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import boa
+
+from scripts.deploy.constants import ZERO_ADDRESS
+from scripts.deploy.utils import deploy_contract
+from settings.config import BASE_DIR, Settings
+
+logger = logging.getLogger(__name__)
+
+
+def deploy_liquidity_gauge_infra(chain_settings: Settings):
```
Comment (amfet42) [inline scripts/deploy/gauge/child_gauge.py:13]: TODO: add post_deploy tests for gauge infra

File: contracts/amm/stableswap/factory/factory_v_100.vy (hunk 1 above)
Comment (bout3fiddy) [reply contracts/amm/stableswap/factory/factory_v_100.vy:137]: you can do it via: commit transfer ownership > accept transfer ownership. i think deployer should only get one chance for setting everything up.

File: scripts/deploy/__init__.py (hunk 2 above)
Comment (bout3fiddy) [reply scripts/deploy/__init__.py:121]: can we do this in a separate PR? i'm not sure if this breaks backend

**Review by amfet42 — APPROVED** [review]

**Pull request merged**